pages/bookmarks.jsx

```diff
@@ -32,7 +33,12 @@ export default React.createClass({
   componentDidMount() {
     // get IDs of user's bookmarked entries
     this.setState({bookmarkedIds: getBookmarks()}, () => {
-      this.fetchData();
+      if (pageSettings.restore) {
+        // restore state back to what is stored in pageSettings
+        this.setState(Object.assign({},this.state,pageSettings.currentList));
```

Comment:
Nice catch! I think originally I had a 3rd argument that later got removed. I forgot to update the rest of the code in that line that's why.